src/main/java/bio/terra/workspace/service/datareference/flight/CreateDataReferenceStep.java

```diff
@@ -57,10 +53,15 @@ public StepResult undoStep(FlightContext flightContext) {
     FlightMap inputMap = flightContext.getInputParameters();
     FlightMap workingMap = flightContext.getWorkingMap();
 
-    if (workingMap.get(CREATE_DATA_REFERENCE_COMPLETED_KEY, Boolean.class)) {
-      UUID referenceId = inputMap.get(DataReferenceFlightMapKeys.REFERENCE_ID, UUID.class);
-      UUID workspaceId = inputMap.get(DataReferenceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    UUID referenceId = workingMap.get(DataReferenceFlightMapKeys.REFERENCE_ID, UUID.class);
+    UUID workspaceId = inputMap.get(DataReferenceFlightMapKeys.WORKSPACE_ID, UUID.class);
+
+    try {
       dataReferenceDao.deleteDataReference(workspaceId, referenceId);
+    } catch (DataAccessException notFoundEx) {
```

Comment:
DataAccessException could be things other than "not found." Anything other than "not found" is still an error. Use a narrower exception type, or consider changing the return type of deleteDataReference if we care about distinguishing between something vs nothing being deleted.